dialogue-serde/src/main/java/com/palantir/conjure/java/dialogue/serde/Encodings.java

```diff
@@ -37,6 +39,9 @@
 
     private Encodings() {}
 
+    private static final Supplier<ObjectMapper> JSON_MAPPER =
```

Comment:
why memoize? can't we just have a static field?

Reply:
imo it's a bit cleaner to avoid work until it's requested, we don't want expensive class loading. When failures occur, it's much nicer to have a stack trace through a call tree than a class init error.

Reply:
In addition to the stacktraces, there's also the perf thing: if everyone did big work in their static fields, then witchcraft startup would get slower and slower.  If we all use suppliers.memoize then the world starts up fast

Reply:
We use a variation of this in an internal project to memoize encoding.serializer/deserializer creation in conjure-undertow, to avoid relatively expensive ObjectReader/ObjectWriter creation until the first time they're needed. Might be worth implementing by default in both dialogue and conjure-undertow.

Reply:
Ok, its pretty ubiquitous to have static mappers. Perhaps we should write an errorprone rule to try to avoid it. IMO its not that big of deal though